records_versioning/signals.py

```
@@ -0,0 +1,92 @@
+    
+    اثرات جانبی: ایجاد یا ثبت یک ورژن از طریق save_with_version. هیچ استثنایی در مسیر افتادن کاربر سیستمی به بیرون پرتاب نمی‌شود (خطای عدم وجود کاربر صامت نادیده گرفته می‌شود).
+    """
+    # Try to get the user from various sources
+    user = None
+    
+    # Check if instance has created_by or updated_by that are User instances
+    if hasattr(instance, 'created_by') and isinstance(instance.created_by, User):
+        user = instance.created_by
+    elif hasattr(instance, 'updated_by') and isinstance(instance.updated_by, User):
+        user = instance.updated_by
+    elif hasattr(instance, 'primary_doctor') and isinstance(instance.primary_doctor, User):
```

Ruff (0.12.2)

Line too long (91 > 88)

(E501)